patches/net/minecraft/client/gui/Gui.java.patch

```diff
@@ -370,15 +370,42 @@
                  this.toolHighlightTimer = (int)(40.0 * this.minecraft.options.notificationDisplayTime().get());
              } else if (this.toolHighlightTimer > 0) {
                  this.toolHighlightTimer--;
-@@ -1287,6 +_,11 @@
-                 p_282761_.drawString(font, SAVING_TEXT, p_282761_.guiWidth() - j - 10, p_282761_.guiHeight() - 15, k);
-             }
+@@ -1289,8 +_,13 @@
          }
-+    }
-+
+     }
+ 
 +    @org.jetbrains.annotations.ApiStatus.Internal
 +    public void initModdedOverlays() {
 +        this.layerManager.initModdedLayers();
-     }
- 
++    }
++
      @OnlyIn(Dist.CLIENT)
+-    public static enum HeartType {
++    public static enum HeartType implements net.neoforged.neoforge.common.IExtensibleEnum {
+         CONTAINER(
+             new ResourceLocation("hud/heart/container"),
+             new ResourceLocation("hud/heart/container_blinking"),
+@@ -1406,8 +_,22 @@
+             } else {
+                 gui$hearttype = NORMAL;
+             }
+-
++            gui$hearttype = net.neoforged.neoforge.event.EventHooks.firePlayerHeartTypeEvent(p_168733_, gui$hearttype);
```

Comment:
Also, thinking about it, instead of reassigning, it may be worth just changing the `return` to `return firePlayerHeartTypeEvent(...)`

Reply:
From a patch perspective the reassignment is better, since changing the return will trigger the following:
```diff
- return gui$hearttype;
+ return net.neoforged.neoforge.event.EventHooks.firePlayerHeartTypeEvent(p_168733_, gui$hearttype);
```

Reply:
Ah, right; it's one addition vs one addition and one removal.